### PR TITLE
integrate fastlane-plugin-update_project_codesigning

### PR DESCRIPTION
**Pull request by hjanuschka**
This PR integrates the `fastlane-plugin_update_project_codesigning`
for not breaking existing setups - we decided to change the name 👍 

cc @mathiasAichinger 🍻 🎉 🎈 

adds the following actions:
  * automatic_code_signing
  * enable_automatic_code_signing ( as alias )
  * disable_automatic_code_signing ( as alias ) 

TODO:
  - [x] add specs (the last 10% are calls to `self.description`, `self.details`, `self.example_code`, `self.return_value`, `self.authors` and `self.is_supported?(platform)` ) 
bildschirmfoto 2017-03-02 um 10 53 24

  - [x] update codesigning guide -> https://github.com/fastlane/docs/pull/297
  - [x] release plugin as new version, with deprecation note https://github.com/hjanuschka/fastlane-plugin-update_project_codesigning/pull/7
  - [x] send me some hugs, as i am loosing my most popular - most downloaded gem 😢

### CODE REVIEW

File: fastlane/lib/fastlane/actions/automatic_code_signing.rb [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        path = params[:path]
+        path = File.join(File.expand_path(path), "project.pbxproj")
+
+        project = Xcodeproj::Project.open(params[:path])
+        UI.user_error!("Could not find path to project config '#{path}'. Pass the path to your project (not workspace)!") unless File.exist?(path)
+        UI.message("Updating the Automatic Codesigning flag to #{params[:use_automatic_signing] ? 'enabled' : 'disabled'} for the given project '#{path}'")
+
+        unless project.root_object.attributes["TargetAttributes"]
+          UI.error("Seems to be a very old project file format")
+          UI.error("PLEASE BACKUP ALL FILES before doing this.")
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/automatic_code_signing.rb]: Don't we want people to upgrade to the new format themselves using Xcode?

File: fastlane/lib/fastlane/actions/automatic_code_signing.rb [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          if ENV["FL_PROJECT_SIGNING_FORCE_UPGRADE"] || UI.confirm("Proceed with upgrade to xcode8 format?")
+            UI.important("Upgrading project to use xcode8 signing stuff")
+            unless params[:team_id]
+              UI.important("TEAM id is not set")
+              UI.error!("Provide :team_id")
+            end
+
+            # set upgrade market to xcdoe8
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/automatic_code_signing.rb]: whops

**Comment by hjanuschka** [comment]
@KrauseFx true point, i removed the auto-project upgrade, and raise a `user_error!`  with a note to upgrade via xcode.

**Comment by mathiasAichinger** [comment]
wohoo 🍻🎉

**Comment by pschneider** [comment]
Just a note:
As a unexperienced user with Code Signing I would love an example integrated into fastlane (disable, run match, gym, ...?)

**Comment by KrauseFx** [comment]
🤗 🤗 🤗 🤗 🤗 🤗  for @hjanuschka 

However, having the first plugin that's integrated into core is a big deal, congrats on that 🚀

**Comment by hjanuschka** [comment]
ok ready for review - i am unsure how to fix CI as it moans about `Could not find method or action named 'enable_automatic_code_signing'`

File: fastlane/lib/fastlane/actions/automatic_code_signing.rb [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+          sett["ProvisioningStyle"] = params[:use_automatic_signing] ? 'Automatic' : 'Manual'
+          if params[:team_id]
+            sett["DevelopmentTeam"] = params[:team_id]
+            UI.important("Set Team id to: #{params[:team_id]} for target: #{found_target[:name]}")
+          end
+          changed_targets << found_target[:name]
+        end
+        project.save
+
+        if changed_targets.length == 0
```
Comment (giginet) [inline fastlane/lib/fastlane/actions/automatic_code_signing.rb]: `empty?`

File: fastlane/lib/fastlane/actions/automatic_code_signing.rb [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+          UI.important("None of the specified targets has been modified")
+          UI.important("available targets:")
+          target_dictionary.each do |tar|
```
Comment (giginet) [inline fastlane/lib/fastlane/actions/automatic_code_signing.rb]: `target`

It should not be dropped.

File: fastlane/lib/fastlane/actions/automatic_code_signing.rb [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            UI.important("\t* #{tar[:name]}")
+          end
+        else
+          UI.success("Successfully updated project settings to use ProvisioningStyle '#{params[:use_automatic_signing] ? 'Automatic' : 'Manual'}'")
+          UI.success("Modified Targets:")
+          changed_targets.each do |tar|
```
Comment (giginet) [inline fastlane/lib/fastlane/actions/automatic_code_signing.rb]: `target`

**Review by giginet — APPROVED** [review]
I'm happy to integrate this plugin 🎉 

I commented some a bit points.

**Comment by hjanuschka** [comment]
@giginet thank you - feedback addressed - @KrauseFx  can you gimme a final review before we merge it ? 👍

**Review by KrauseFx — APPROVED** [review]
:rocket:

**Comment by KrauseFx** [comment]
@hjanuschka Is this ready to go? We can merge now, and do a release tomorrow morning 🚀

Seems like the tests are failing due to the code samples provided

```

[11:49:15]: Found errors in the documentation, more information above
[11:49:15]: Error found in code sample:
# enable automatic code signing
enable_automatic_code_signing
Could not find method or action named 'enable_automatic_code_signing'
[11:49:15]: Error found in code sample:
enable_automatic_code_signing(
  path: "demo-project/demo/demo.xcodeproj"
)
Could not find method or action named 'enable_automatic_code_signing'
[11:49:15]: Error found in code sample:
# disable automatic code signing
disable_automatic_code_signing
Could not find method or action named 'disable_automatic_code_signing'
[11:49:15]: Error found in code sample:
disable_automatic_code_signing(
  path: "demo-project/demo/demo.xcodeproj"
)
Could not find method or action named 'disable_automatic_code_signing'
[11:49:15]: Error found in code sample:
# also set team id
disable_automatic_code_signing(
  path: "demo-project/demo/demo.xcodeproj",
  team_id: "XXXX"
)
Could not find method or action named 'disable_automatic_code_signing'
[11:49:15]: Error found in code sample:
# Only specific targets
  disable_automatic_code_signing(
    path: "demo-project/demo/demo.xcodeproj",
    use_automatic_signing: false,
    targets: ["demo"]
  )

Could not find method or action named 'disable_automatic_code_signing'
[11:49:15]: Syntax error in code sample:
 # via generic action
automatic_code_signing(
  path: "demo-project/demo/demo.xcodeproj"
  use_automatic_signing: false
)
(eval):4: syntax error, unexpected tIDENTIFIER, expecting ')'
  use_automatic_signing: false
                       ^
```

**Comment by KrauseFx** [comment]
Fixes https://github.com/fastlane/fastlane/issues/8477

**Review by KrauseFx — APPROVED** [review]

**Comment by janpio** [comment]
@hjanuschka Sorry for hijacking this old issue, but: 
Is this "Seems to be a very old project file format" logic that was removed when integrating this plugin available as a standalone plugin somewhere? I can't really open the project in Xcode to get the files updated, as this should all be really automated :/

**Comment by janpio** [comment]
Sorry for the notification, I got it "working" by just stealing the code you removed into a lane in my Fastfile for now.

@KrauseFx you wrote "Don't we want people to upgrade to the new format themselves using Xcode?" which initated the removal of this code. Is this some deeper position or did you just prefer not to have this bundled with this action?

**Comment by KrauseFx** [comment]
Hey @janpio, please create a new issue so we can properly track it :)

**Pull request merged**